### ISSUE TITLE
Integrate github action

### ISSUE THREAD

**Issue by hennersz** [nix-community/nixbox#89]
Heya I've managed to get nixbox building in github actions for virtualbox and libvirt in [my own fork](https://github.com/hennersz/nixbox)
They are currently published under my own namespace in vagrant cloud.
If the maintainers of this repo want to integrate it and publish under the nixos account I can open a PR

**Comment by pdietl** [comment]
@zimbatm are you interesting in adding such a github action? Maybe to trigger on a release? Or would you be down to upload new boxes for nix 23.11? I can build the boxes of all the providers, but obviously I can't upload them.

**Comment by hennersz** [comment]
Fyi the actions started intermittently failing more and more often, so I need to do some work to try and fix it

**Comment by zimbatm** [comment]
yeah sounds good. especially now that GHA supports nested virtualization, it should make building those VM images much easier. then upload them as release assets.

I'm not actively using this project anymore, so if any of you or @ifurther also wants to take over maintenance, that would be nice.

**Comment by ifurther** [comment]
I am interesting this, maybe we can do it.

**Comment by zimbatm** [comment]
Invited you both since you are active on the project. Make yourself at home and let me know if you need anything.

**Comment by pdietl** [comment]
Thanks @zimbatm! How do we get the credentials for uploading to VagrantCloud?

**Comment by ifurther** [comment]
Maybe should create the Organization in VagrantCloud.

**Comment by zimbatm** [comment]
I added the ATLAS_TOKEN secret to GitHub Actions, for the automation. If you sign up to https://app.vagrantup.com/account/new and give me your account, I will also give you access to the org.

**Comment by ifurther** [comment]
@zimbatm my vargrnat username is same for my github username.
 @pdietl the github action image have the packer, but no VirtualBox.

**Comment by zimbatm** [comment]
Invited you to the "nixbox" org. I propose that we start fresh, so you can also organize things like you want.

**Comment by ifurther** [comment]
Thanks @zimbatm I will update makefile

**Comment by ifurther** [comment]
@zimbatm I push the action for building the image, but it can't create the `nixbox/nixos` box. Maybe should you create it.

**Comment by ifurther** [comment]
@zimbatm I can't set the nixbox/nixos to public

**Comment by zimbatm** [comment]
https://app.vagrantup.com/nixbox/boxes/nixos is public isn't it?

**Comment by ifurther** [comment]
Now is public, thanks

**Comment by ifurther** [comment]
Hi @hennersz I add the ci building the image now, cloud you help me add the flake function to new ?

**Comment by hennersz** [comment]
might be able to look at this over the weekend
